### PR TITLE
Ignore BigDecimal and BigInteger reflection hierarchy warnings

### LINKED ISSUE #9189: BigDecimal might not be automatically registered for reflection
**Describe the bug**
When starting a Quarkus application, it displays the warning:
Unable to properly register the hierarchy of the following classes for reflection as they are not in the Jandex index:
    - java.math.BigDecimal

**Expected behavior**
No warning should be displayed and BigDecimal should be automatically registered for reflection.

**Actual behavior**
The warning is displayed and (need to check this) the BigDecimal is not automatically registered for reflection.

**To Reproduce**
A simple Quarkus application with just two classes are enough to reproduce it. It is here 
https://stackoverflow.com/questions/61659029/quarkus-warning-about-reflection-of-bigdecimal

**Environment (please complete the following information):**
Quarkus 1.4.2.Final. It does not happen on Quarkus 1.4.1.Final

### PR DESCRIPTION
**Pull request by geoand**
These classes are never part of the Jandex index so there
is no point in warning about them.

Fixes: #9189

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/deployment/steps/ReflectiveHierarchyStep.java [hunk 1]
```diff
@@ -82,9 +74,26 @@ public void build() throws Exception {
         }
     }
 
-    private void addReflectiveHierarchy(ReflectiveHierarchyBuildItem reflectiveHierarchyBuildItem, Type type,
+    private void removeIgnored(Set<DotName> unindexedClasses, List<ReflectiveHierarchyIgnoreWarningBuildItem> ignored) {
+        if (ignored.isEmpty()) {
+            return;
+        }
+        Set<DotName> ignoredWarnings = new HashSet<>();
+        // if at least one of the predicates indicates that warning should be ignored, then the ensure that it is ignored
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/steps/ReflectiveHierarchyStep.java:82]: `then the ensure`?

File: core/deployment/src/main/java/io/quarkus/deployment/steps/ReflectiveHierarchyStep.java [hunk 2]
```diff
@@ -36,23 +34,17 @@
...
-    CombinedIndexBuildItem combinedIndexBuildItem;
-
-    @Inject
-    BuildProducer<ReflectiveClassBuildItem> reflectiveClass;
-
-    @Inject
-    List<ReflectiveHierarchyIgnoreWarningBuildItem> ignored;
-
-    @Inject
-    List<ReflectiveClassFinalFieldsWritablePredicateBuildItem> finalFieldsWritablePredicates;
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/steps/ReflectiveHierarchyStep.java:52]: Please do that in a separate commit next time. It makes the review far harder than it needs to be.

**Review by gsmet — CHANGES_REQUESTED** [review]
Spotted a typo. Looks good otherwise but the commit is a bit hard to follow so fingers crossed.

File: core/deployment/src/main/java/io/quarkus/deployment/steps/ReflectiveHierarchyStep.java (hunk 2 above)
Comment (geoand) [reply core/deployment/src/main/java/io/quarkus/deployment/steps/ReflectiveHierarchyStep.java:52]: You are absolutely right. I thought of it myself, but I guess I decided against it because I (stupidly) decided it would be best to economize on commits :laughing:

File: core/deployment/src/main/java/io/quarkus/deployment/steps/ReflectiveHierarchyStep.java (hunk 1 above)
Comment (geoand) [reply core/deployment/src/main/java/io/quarkus/deployment/steps/ReflectiveHierarchyStep.java:82]: Fixed

**Review by gsmet — APPROVED** [review]
